### ISSUE TITLE
[SYCL] Compiler crashes with no input files

### ISSUE THREAD

**Issue by j-stephan** [intel/llvm#470]
Pulling this upstream from triSYCL/sycl#62. The command line

```
clang++ -fsycl foo.cpp -lbla
```

will produce a segmentation fault if the compiler can't find `foo.cpp`. This only occurs when trying to link a library, the command line without `-lbla` will just tell me that there is no `foo.cpp`.

Error message:

```
clang-9: error: no such file or directory: 'foo.cpp'
Stack dump:
0.	Program arguments: clang++ -std=c++2a -fsycl foo.cpp -lbla 
1.	Compilation construction
2.	Building compilation jobs
3.	Building compilation jobs
4.	Building compilation jobs
5.	Building compilation jobs
6.	Computing output path
 #0 0x000055e4f76ec5ca llvm::sys::PrintStackTrace(llvm::raw_ostream&) (/home/jan/software/sycl/intel/bin/clang-9+0x27245ca)
 #1 0x000055e4f76ea2d4 llvm::sys::RunSignalHandlers() (/home/jan/software/sycl/intel/bin/clang-9+0x27222d4)
 #2 0x000055e4f76ea455 SignalHandler(int) (/home/jan/software/sycl/intel/bin/clang-9+0x2722455)
 #3 0x00007f802f9c3f40 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x13f40)
 #4 0x00007f802f5af0c7 /build/glibc-KRRWSm/glibc-2.29/string/../sysdeps/x86_64/multiarch/strlen-avx2.S:96:0
 #5 0x000055e4f7c6c047 clang::driver::Driver::GetNamedOutputPath(clang::driver::Compilation&, clang::driver::JobAction const&, char const*, llvm::StringRef, bool, bool, llvm::StringRef) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca4047)
 #6 0x000055e4f7c70308 clang::driver::Driver::BuildJobsForActionNoCache(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca8308)
 #7 0x000055e4f7c71167 clang::driver::Driver::BuildJobsForAction(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca9167)
 #8 0x000055e4f7c6f206 clang::driver::Driver::BuildJobsForActionNoCache(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca7206)
 #9 0x000055e4f7c71167 clang::driver::Driver::BuildJobsForAction(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca9167)
#10 0x000055e4f7c73804 void llvm::function_ref<void (clang::driver::Action*, clang::driver::ToolChain const*, char const*)>::callback_fn<clang::driver::Driver::BuildJobsForActionNoCache(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const::'lambda0'(clang::driver::Action*, clang::driver::ToolChain const*, char const*)>(long, clang::driver::Action*, clang::driver::ToolChain const*, char const*) (/home/jan/software/sycl/intel/bin/clang-9+0x2cab804)
#11 0x000055e4f7d74394 clang::driver::OffloadAction::doOnEachDeviceDependence(llvm::function_ref<void (clang::driver::Action*, clang::driver::ToolChain const*, char const*)> const&) const (/home/jan/software/sycl/intel/bin/clang-9+0x2dac394)
#12 0x000055e4f7c6ea2a clang::driver::Driver::BuildJobsForActionNoCache(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca6a2a)
#13 0x000055e4f7c71167 clang::driver::Driver::BuildJobsForAction(clang::driver::Compilation&, clang::driver::Action const*, clang::driver::ToolChain const*, llvm::StringRef, bool, bool, char const*, std::map<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, clang::driver::InputInfo, std::less<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > >, std::allocator<std::pair<std::pair<clang::driver::Action const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > const, clang::driver::InputInfo> > >&, clang::driver::Action::OffloadKind) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca9167)
#14 0x000055e4f7c716cb clang::driver::Driver::BuildJobs(clang::driver::Compilation&) const (/home/jan/software/sycl/intel/bin/clang-9+0x2ca96cb)
#15 0x000055e4f7c730fd clang::driver::Driver::BuildCompilation(llvm::ArrayRef<char const*>) (/home/jan/software/sycl/intel/bin/clang-9+0x2cab0fd)
#16 0x000055e4f5bbc4fc main (/home/jan/software/sycl/intel/bin/clang-9+0xbf44fc)
#17 0x00007f802f44fb6b __libc_start_main /build/glibc-KRRWSm/glibc-2.29/csu/../csu/libc-start.c:342:3
#18 0x000055e4f5c4714a _start (/home/jan/software/sycl/intel/bin/clang-9+0xc7f14a)
[1]    4789 segmentation fault (core dumped)  clang++ -std=c++2a -fsycl foo.cpp -lbla
```

**Comment by AlexeySachkov** [comment]
@j-stephan, this issue should be fixed by intel/llvm#495, could you please check?

**Comment by j-stephan** [comment]
Fix confirmed, closing.
